scripts/snakemake_adage_two_level.R

```diff
@@ -0,0 +1,202 @@
+                 'SPU233_M', 'SPU239_M', 'SPU243_M')
+} else {
+  print("comparison group not known")
+}
+
+ 
+# run ADAGE ---------------------------------------------------------------
+
+
+# Notes
+# PAcompendium, eADAGEmodel, and probedistribution are objects that come with the ADAGEpath package.
+
+
+# Read in normalized and filtered compendium that contains samples of interest.
+# Select samples of interest that will be compared with ADAGE.
+# The compendium that is read in has been filtered to remove samples with a high
+# number of unobserved genes (zeroes), or low expression of house keeping genes.
+# The counts are otherwise raw (e.g. produced by salmon), and so are normalized
+# with using the ADAGEpath functions
+rnaseq_data <- read_csv(snakemake@input[['num_reads']], show_col_types = F) %>%
+#rnaseq_data <- read_csv("outputs/filt_norm_compendia/pao1_aligned_compendium_p2_filtered_num_reads.csv", show_col_types = F) %>%
+  dplyr::rename(geneID = `...1`) %>%                            # reset index (no colname) to geneID
+  dplyr::select(geneID, one_of(control_set), one_of(group_set)) # select control samples, experimental samples
+
+# note that some of the group_set may not be in the RNAseq data because those samples 
+# did not pass filtering thresholds when the compendium was originally built
+
+data_raw <- load_dataset(input        = rnaseq_data,
+                         isProcessed  = TRUE,
+                         isRNAseq     = TRUE,
+                         model        = eADAGEmodel,
+                         compendium   = PAcompendium,
+                         quantile_ref = probedistribution,
+                         norm01       = FALSE)
+
+data_normed <- zeroone_norm(input_data = data_raw, use_ref = TRUE, ref_data = PAcompendium)
+
+data_activity <- calculate_activity(input_data = data_normed, model = eADAGEmodel)
+write_tsv(data_activity, snakemake@output[['data_activity']])
+
+# build vector for phenotype information. 
+# use control set and the final number of columns in the rna_seq data to build this information.
+data_pheno <- c(rep("control", length(control_set)), rep(group, (ncol(rnaseq_data) - (length(control_set) + 1))))
+
+limma_result <- build_limma(data_activity,
+                            phenotypes = data_pheno,
+                            control_pheno = "control",
+                            use.bonferroni = TRUE)
+write_tsv(limma_result, snakemake@output[['limma_result']])
+
+active_sigs <- get_active_signatures(limma_result = limma_result,
+                                     pheno_group  = "both",
+                                     method       = "pareto",
+                                     N_fronts     = 10) # adjust N fronts to change the number of returned nodes
+
+
+# plots -------------------------------------------------------------------
+# uncomment these lines to produce default visualizations from ADAGE
```

Comment:
Do you want them to start commented then? Make it an opt-in feature instead?

Reply:
woops, good catch! I'll remove this line of comment
```suggestion
```